gunagala/tests/test_imager.py

```diff
@@ -735,6 +736,20 @@ def test_snr_vs_mag(imager, filter_name, tmpdir):
     assert mags3[-1].value == pytest.approx(mags[-1].value - 2.5, abs=0.1)
 
 
+def test_get_pixel_coords(imager):
+    test_coord_string = "189.9976325 -11.6230544"
+
+    # test if someone calls get_pixel_coords without first running set_WCS_centre
+    with pytest.raises(ValueError):
+        imager.get_pixel_coords()
+
+    # now set WCS centre first, then try and get_pixel_coords
+    imager.set_WCS_centre(test_coord_string)
+    centre_field_pixels = imager.get_pixel_coords()
+    assert imager.wcs._naxis1 == centre_field_pixels.shape[1]
+    assert imager.wcs._naxis2 == centre_field_pixels.shape[0]
```

Comment:
James Synge would complain about these tests being non-hermetic, i.e. their success depends on being run in the right order. Ideally rather than using the `imager` fixture for these (which has `module` scope and so gives the same `Imager` object to each test) you should create a fresh `Imager` instance to be sure that `CRVAL` hasn't been set yet. Otherwise these tests could fail in the future simply because some other test that sets CRVAL gets added further up the file.